internal/transport/transport_test.go

```diff
@@ -2029,10 +2029,11 @@ func (s) TestClientDecodeHeaderStatusErr(t *testing.T) {
 			metaHeaderFrame: &http2.MetaHeadersFrame{
 				Fields: []hpack.HeaderField{
 					{Name: "content-type", Value: "application/json"},
+					{Name: ":status", Value: "200"},
 				},
 			},
 			wantStatus: status.New(
-				codes.Internal,
+				codes.Unknown,
```

Comment:
What is the purpose of this test change?  This doesn't test the modified code AFAICT.

Can this test case list handle a new case that sets the end stream flag in the headers frame?  If so, we should be able to verify that a missing status (with all other requirements met) is detected in that scenario.  This is the case that would have failed before the change made above.

Reply:
I reverted the test change - the message was different and I'm not sure what I was thinking. I had the tests originally set to add the end stream flag, if no `HeadersFrame` was present but I think that might have made the tests a little less explicit. Let me know what you think.

I did add a test explicitly for testing the http status missing from the headers.